src/engine_core/rendering/src/Vulkan/VkTypes.hpp

```diff
@@ -9,19 +9,34 @@
 #include <vulkan/vulkan.h>
 #include <magic_enum.hpp>
 #include <glm/vec4.hpp>
+#include <glm/mat4x4.hpp>
 
 
 // Stuff from vk_mem_alloc to avoid cyclical references
 struct VmaAllocation_T;
 using VmaAllocation = VmaAllocation_T *;
 
+struct Vertex {
+	glm::vec3 position;
+	float uv_x = 0.0f;
+	glm::vec3 normal = { 1, 0, 0 };
+	float uv_y = 0.0f;
+	glm::vec4 color = glm::vec4{ 1.f };
+};
+
+// push constants for our mesh object draws
+struct GPUDrawPushConstants {
+	glm::mat4 worldMatrix;
+	VkDeviceAddress vertexBuffer;
+};
+
 struct AllocatedImage
 {
-    VkImage image;
-    VkImageView imageView;
-    VmaAllocation allocation;
-    VkExtent3D imageExtent;
-    VkFormat imageFormat;
+	VkImage image;
```

Comment:
Is it okay to not initialize it?
If not, I suggest creating a constructor and deleting the default one